core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -414,13 +414,6 @@ private static BulkItemResultHolder executeUpdateRequest(UpdateRequest updateReq
         FAILURE
     }
 
-    static {
-        assert Version.CURRENT.minimumCompatibilityVersion().after(Version.V_5_0_0) == false:
```

Comment:
I did some research and I think I get why this assertion was put in there. This assertion should check for V_6_0_0. That said, I'm not even sure we should have it and the accompanying TODO. I suggest changing the assertion to V_6_0_0 in this PR and I will work with @dakrone to clarify this in a follow up.